fem/fespace.cpp

```diff
@@ -1788,59 +1828,66 @@ void FiniteElementSpace::GetBdrElementDofs(int i, Array<int> &dofs) const
 
 void FiniteElementSpace::GetFaceDofs(int i, Array<int> &dofs) const
 {
-   int j, k, nv, ne, nf, nd, dim = mesh->Dimension();
-   Array<int> V, E, Eo;
-   const int *ind;
-
-   // for 1D, 2D and 3D faces
-   nv = fec->DofForGeometry(Geometry::POINT);
-   ne = (dim > 1) ? fec->DofForGeometry(Geometry::SEGMENT) : 0;
-   if (nv > 0)
-   {
-      mesh->GetFaceVertices(i, V);
-   }
-   if (ne > 0)
+   if (face_dof)
```

Comment:
Perhaps this is more a naming/documentation issue, but reading this code is confusing for me.

The `if (face_dof)` branch is to treat the case of NURBS face dofs separately. However, a reader of the code (who may not be aware that the protected member data `face_dof` is only non-NULL for NURBS meshes) may not make the connection with NURBS meshes.

A comment could help here, or perhaps adding a member function with a more descriptive name (e.g. mentioning NURBS) to check if `face_dof` is not NULL.

Reply:
In the future other fespace types might use the `face_dof` table, so I think linking it to hard with the NURBS case is not necessary. I would argue the current format is quite OO in the sense that NURBS or not is "hidden" and irrelevant. The only relevant question here is, I want a face dof, can I use the `face_dof` table or not. Moreover the current implementation is 100% analogous with `GetElementDofs` and `GetElementBdrDofs`.